python/cudf/cudf/core/dataframe.py

```diff
@@ -704,8 +704,9 @@ def values(self):
 
     def __array__(self, dtype=None):
         raise NotImplementedError(
```

Comment:
`NotImplementedError` gives the impression that the method *will* be implemented at some point:

https://docs.python.org/3/library/exceptions.html#NotImplementedError

Let's raise a `TypeError` like CuPy?

Reply:
Changed to `TypeError`